frame/support/procedural/src/pallet/parse/event.rs

```diff
@@ -0,0 +1,248 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::helper;
+use syn::spanned::Spanned;
+use quote::ToTokens;
+use frame_support_procedural_tools::clean_type_string;
+
+/// List of additional token to be used for parsing.
+mod keyword {
+	syn::custom_keyword!(metadata);
+	syn::custom_keyword!(Event);
+	syn::custom_keyword!(pallet);
+	syn::custom_keyword!(generate_deposit);
+	syn::custom_keyword!(deposit_event);
+}
+
+/// Definition for pallet event enum.
+pub struct EventDef {
+	/// The index of event item in pallet module.
+	pub index: usize,
+	/// The keyword Event used (contains span).
+	pub event: keyword::Event,
+	/// Event metadatas: `(name, args, docs)`.
+	pub metadata: Vec<(syn::Ident, Vec<String>, Vec<syn::Lit>)>,
+	/// A set of usage of instance, must be check for consistency with trait.
+	pub instances: Vec<helper::InstanceUsage>,
+	/// If event is declared with instance.
+	pub has_instance: bool,
+	/// If event is declared with generics.
+	pub is_generic: bool,
+	/// Whether the function `deposit_event` must be generated.
+	pub deposit_event: Option<(syn::Visibility, proc_macro2::Span)>,
+	/// Where clause used in event definition.
+	pub where_clause: Option<syn::WhereClause>,
+}
+
+impl EventDef {
+	/// Return the generic to be used when using Event type
+	///
+	/// Depending on its definition it can be: ``, `T` or `T, I`
+	pub fn event_use_gen(&self) -> proc_macro2::TokenStream {
+		if self.is_generic {
+			if self.has_instance {
+				quote::quote!(T, I)
+			} else {
+				quote::quote!(T)
+			}
+		} else {
+			quote::quote!()
+		}
+	}
+
+	/// Return the generic to be used in `impl<..>` when implementing on Event type.
+	pub fn event_impl_gen(&self) -> proc_macro2::TokenStream {
+		if self.is_generic {
+			if self.has_instance {
+				quote::quote!(T: Config<I>, I)
```

Comment:
```suggestion
				quote::quote!(T: Config<I>, I: 'static)
```

e.g in /frame/support/test/tests/pallet_instance.rs we lose the `'static` bound on the `I` in `Event`

Reply:
Yes you are correct